frontend/src/services/Controller.ts

```diff
@@ -135,7 +137,9 @@ function getEventHandlers() {
     dataReady: (result: boolean) => backend.set({ dataReady: result }),
 
     environment: (result: ILookup<any>) => {
-      backend.set({ environment: result })
+      const state = store.getState()
+      const { backendSetting } = state.backend.environment
+      backend.set({ environment: { ...result, backendSetting } })
```

Comment:
Hmmm - maybe to avoid this and having to keep the default values in the override setting we can leave the override setting as empty '' and just do the override in the actual api calls...